milestone3/frontend1/src/recipePage.jsx

```diff
@@ -1,35 +1,65 @@
 import "./recipePage.css";
-import React from "react";
+import React, { useState, useEffect } from 'react';
 import Navbar from "./navBar";
 
 export default function RecipePage({
   name,
   description,
   imageSrc,
   ingredientList,
-  instructionList,
+  instructions,
   link,
   courtesyOf,
 }) {
   const [newIngredient, setNewIngredient] = React.useState("");
+  const [addIngredient, setAddIngredient] = useState("");
   const [newInstruction, setNewInstruction] = React.useState("");
+  const [addInstruction, setAddInstrction] = useState("");
+
+  useEffect(() => {
+    if (addIngredient === '') return;
+
+    const requestOptions = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ addIngredient: addIngredient })
+    };
+    const updateIngredients = async () => {
+      fetch(`http://localhost:3001/api/recipe/${name}/ingredient`, requestOptions)
+    }
+    updateIngredients();
+  }, [addIngredient])
+
+  useEffect(() => {
+    if (addInstruction === '') return;
+
+    const requestOptions = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ newInstruction: addInstruction })
+    };
+    const updateInstrcutions = async () => {
+      fetch(`http://localhost:3001/api/recipe/${name}/instruction`, requestOptions)
+    }
+    updateInstrcutions();
+  }, [addInstruction])
```

Comment:
You don't need to have a useEffect for POST requests. Remember, By using the useEffect Hook, you tell React that your component needs to do something after render.

However, the POST request should be ran whenever the addIngredient or addInstruction button is clicked. (But it looks like you got it working this way so whatever works works!!)